devito/ir/equations/algorithms.py

```diff
@@ -135,8 +136,13 @@ def _lower_exprs(expressions, subs):
             if dimension_map:
                 indices = [j.xreplace(dimension_map) for j in indices]
 
-            mapper[i] = f.indexed[indices]
+            # Handle Array
```

Comment:
@FabioLuporini There is probably a cleaner way but don't have time to spend more on this rn